portals/admin/src/main/webapp/source/src/app/components/Workflow/APIRevisionDeployment.jsx

```diff
@@ -0,0 +1,528 @@
+import IconButton from '@material-ui/core/IconButton';
+import { makeStyles } from '@material-ui/core/styles';
+import SearchIcon from '@material-ui/icons/Search';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import Card from '@material-ui/core/Card';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import MUIDataTable from 'mui-datatables';
+import ContentBase from 'AppComponents/AdminPages/Addons/ContentBase';
+import InlineProgress from 'AppComponents/AdminPages/Addons/InlineProgress';
+import Alert from 'AppComponents/Shared/Alert';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import HelpBase from 'AppComponents/AdminPages/Addons/HelpBase';
+import DescriptionIcon from '@material-ui/icons/Description';
+import Link from '@material-ui/core/Link';
+import Configurations from 'Config';
+import API from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import * as dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import localizedFormat from 'dayjs/plugin/localizedFormat';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import Box from '@material-ui/core/Box';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import WarningBase from 'AppComponents/AdminPages/Addons/WarningBase';
+import { Alert as MUIAlert } from '@material-ui/lab';
+
+const useStyles = makeStyles((theme) => ({
+    searchInput: {
+        fontSize: theme.typography.fontSize,
+    },
+    block: {
+        display: 'block',
+    },
+    contentWrapper: {
+        margin: theme.spacing(2),
+    },
+    approveButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.success.light,
+    },
+    rejectButton: {
+        textDecoration: 'none',
+        backgroundColor: theme.palette.error.light,
+    },
+}));
+
+/**
+ * Render a list
+ * @param {JSON} props props passed from parent
+ * @returns {JSX} Header AppBar components.
+ */
+function ListLabels() {
+    const intl = useIntl();
+    const [data, setData] = useState(null);
+    const restApi = new API();
+    const classes = useStyles();
+    const [searchText, setSearchText] = useState('');
+    const [isUpdating, setIsUpdating] = useState(null);
+    const [buttonValue, setButtonValue] = useState();
+    const [hasListPermission, setHasListPermission] = useState(true);
+    const [errorMessage, setError] = useState(null);
+
+    /**
+     * API call to get Detected Data
+     * @returns {Promise}.
+     */
+    function apiCall() {
+        return restApi
+            .workflowsGet('AM_REVISION_DEPLOYMENT')
+            .then((result) => {
+                const workflowlist = result.body.list.map((obj) => {
```

Comment:
Can obj and obj.properties be null?

Reply:
This obj would not be empty as if there are no revisions the results body is empty and if obj are available they always have properties.